module/VuFindAdmin/src/VuFindAdmin/Controller/FeedbackController.php

```diff
@@ -0,0 +1,306 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Class FeedbackController
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Moravian Library 2022.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  VuFindAdmin\Controller
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+namespace VuFindAdmin\Controller;
+
+use Laminas\Db\Sql\Select;
+use VuFind\Db\Table\Feedback;
+
+/**
+ * Class FeedbackController
+ *
+ * @category VuFind
+ * @package  VuFindAdmin\Controller
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+class FeedbackController extends AbstractAdmin
+{
+    /**
+     * Params
+     *
+     * @var array
+     */
+    protected $params;
+
+    /**
+     * Get the url parameters
+     *
+     * @param string $param A key to check the url params for
+     *
+     * @return string
+     */
+    protected function getParam($param)
+    {
+        return $this->params[$param] ?? $this->params()->fromPost(
+            $param,
+            $this->params()->fromQuery($param, null)
+        );
+    }
+
+    /**
+     * Home action
+     *
+     * @return \Laminas\View\Model\ViewModel
+     */
+    public function homeAction()
+    {
+        $this->params = $this->params()->fromQuery();
+        $view = $this->createViewModel();
+        $view->setTemplate('admin/feedback/home');
+        $feedbackTable = $this->getFeedbackTable();
+        $feedback = $feedbackTable->getFeedbackByFilter(
+            $this->convertFilter($this->getParam('form_name')),
+            $this->convertFilter($this->getParam('site_url')),
+            $this->convertFilter($this->getParam('status'))
+        );
+        $view->setVariables(
```

Comment:
What if you just call createViewModel here with this array, instead of creating the view earlier and then setting the variables separately? You would just have to move down the setTemplate call above, and then the code is simplified by a line.

Reply:
You are right. Done